src/main/java/com/microsoft/azure/vmagent/remote/AzureVMAgentSSHLauncher.java

```diff
@@ -104,6 +104,12 @@ public void launch(SlaveComputer agentComputer, TaskListener listener) {
         boolean successful = false;
         Session session = null;
 
+        SlaveComputer slaveComputer = agent.getComputer();
+        if (slaveComputer == null) {
+            LOGGER.log(Level.SEVERE, "AzureVMAgentSSHLauncher: launch: Got null computer.");
+            handleLaunchFailure(agent, Constants.AGENT_POST_PROV_AUTH_FAIL);
```

Comment:
Should change failure message here, not a auth failure.

Reply:
I've pushed another message, feel free to amend it as you wish